src/transaction/sign.js

```diff
@@ -45,4 +42,17 @@ function sign(txJSON: string, secret: string, options: Object = {}
   }
 }
 
+function sign(txJSON: string, secret?: any, options?: Object, keypair?: Object
+): {signedTransaction: string; id: string} {
+  if(typeof(secret) === 'string') {
+    // we can't validate that the secret matches the account because
```

Comment:
I'd say "we don't..." rather than "can't".

Wondering aloud, is there any reason to validate that the publicKey and privateKey match each other?

Reply:
👍  I think it would only be useful in a world where the ledger would only accept a signature from the secret that matches the account. However, there can be other keys that could sign for the account, I don't think there's a reason to validate here.